Make run_started_at timezone aware

mturzanska opened the pull request:
Addresses  #474 

Set run_started_at timezone to UTC
Enable timezone change in models
Extend requirements
Extend tests

drewbanin commented:
@mturzanska this is great! You can ignore the failing circle CI check -- we'll re-open this PR to run it with snowflake / bigquery credentials.

I have a couple of tiny comments / thoughts, but also keen to get @cmcarthur's input here. Thanks for submitting!!

drewbanin commented on dbt/context/common.py:
```diff
@@ -279,6 +280,7 @@ def generate(model, project, flat_graph, provider=None):
         "model": model,
         "post_hooks": post_hooks,
         "pre_hooks": pre_hooks,
+        "pytz": pytz,
```
`pytz` is the first module that we're adding to the dbt context, but conceivably we'd want to add many more like `datetime`, `collections` etc. That makes me think we should namespace all of these modules under a `libraries` or `modules` or `imports` name, so the namespace doesn't get super polluted.

```
"modules": {
  "pytz" : pytz,
  "datetime": datetime,
  ....
}
```

@mturzanska  and @cmcarthur what do you think about that?

cmcarthur replied:
namespacing these is a good idea. `modules` seems like a perfectly good name for that namespace.

let's just make sure the syntax remains nice, i.e. (`modules.pytz.timezone("America/New_York")`). @mturzanska if that syntax doesn't work after updating this, you can make `modules` an `AttrDict`:

```
...
"modules": AttrDict({
  "pytz": pytz
})
...
```

should do the trick.

drewbanin commented on requirements.txt:
```diff
@@ -1,5 +1,7 @@
 argparse>=1.2.1
+freezegun==0.3.9
 Jinja2>=2.8
+pytz==2017.2
```
i _think_ these modules also need to be added to `setup.py`

cmcarthur replied:
they do

drewbanin commented on test/integration/025_timezones_test/models/timezones.sql:
```diff
@@ -0,0 +1,10 @@
+}}
+
+select
+    '{{ run_started_at.astimezone(pytz.timezone("America/New_York")) }}' as run_started_at_est,
```
:+1:

cmcarthur commented:
added my comments. @mturzanska this looks great, thanks for contributing!

mturzanska commented:
Addressed all comments but let me know if there is anything missing still.
Other than that I will have a look at the list of issues to grab sth else.

drewbanin commented:
@mturzanska we're going to try to get Circle to run the all of the integration tests here, then we should be good to get this merged in :)

drewbanin commented:
ok, i ran these locally and everything passed. As a side note, @cmcarthur and I are going to investigate how to make these tests run correctly for external contributors.

@mturzanska really good work, merging this now. It's going into an `0.9.1` branch and will be released after our `0.9.0` changes are out of alpha! We will let you know when it's live :)
